ale/base/type_sensor.py

```diff
@@ -362,3 +362,63 @@ def line_times(self):
         : array
         """
         raise NotImplementedError
+
+
+class Cavhor():
+    """
+    Mixin for largely ground based sensors to add the an
+    extra step in the frame chain to go from ground to J2000
+    """
+
+    @property
+    def cahvor_rotation_matrix(self):
+        """
+        Computes the cahvor rotation matrix for the instrument to Rover frame
+        
+        Returns
+        -------
+        : array
+          Rotation Matrix as a 2D numpy array
+        """
+        keys = ['GEOMETRIC_CAMERA_MODEL', 'GEOMETRIC_CAMERA_MODEL_PARMS']
+        for key in keys:
+            camera_model_group = self.label.get(key, None)
+            if camera_model_group is not None:
+                break
+        if camera_model_group is None:
+            return []
+        A = np.array(camera_model_group["MODEL_COMPONENT_2"])
+        H = np.array(camera_model_group["MODEL_COMPONENT_3"])
+        V = np.array(camera_model_group["MODEL_COMPONENT_4"])
+        h_c = np.dot(A, H)
+        h_s = np.linalg.norm(np.cross(A, H))
+        v_c = np.dot(A, V)
+        v_s = np.linalg.norm(np.cross(A, V))
+        H_prime = (H - h_c * A)/h_s
+        V_prime = (V - v_c * A)/v_s
+        rot_matrix = np.array([H_prime, -V_prime, -A])
+
+        return rot_matrix
+
+    @property
+    def frame_chain(self):
+        """
+        Returns a modified frame chain with the cahvor models extra rotation
+        added into the model
+
+        Returns
+        -------
+        : object
+          A networkx frame chain object
+        """
+        frame_chain = super().frame_chain
+        cahvor_quats = np.zeros(4)
+        cahvor_quat_from_rotation = spice.m2q(self.cahvor_rotation_matrix)
+        cahvor_quats[:3] = cahvor_quat_from_rotation[1:]
+        cahvor_quats[3] = cahvor_quat_from_rotation[0]
+        site_frame = str(self.label["GEOMETRIC_CAMERA_MODEL_PARMS"]["REFERENCE_COORD_SYSTEM_INDEX"][0])
+        site_frame = spice.bods2c("MSL_SITE_" + site_frame)
+        rover_frame = spice.bods2c(self.instrument_id)
```

Comment:
I tried `spice.bds2c("MSL_SITE_62")` and get "Spice returns not found". I also get this error for `spice.bods2c(self.instrument_id)` or `spice.bods2c("MSL_MASTCAM_RIGHT")` based on https://naif.jpl.nasa.gov/pub/naif/MSL/kernels/fk/msl_v06.tf. 

I ended having to just manually set the rover frame in ikid() in the driver.